test/object-store/sync/sync_test_utils.cpp

```diff
@@ -361,6 +361,34 @@ void wait_for_object_to_persist_to_atlas(std::shared_ptr<SyncUser> user, const A
         std::chrono::minutes(15), std::chrono::milliseconds(500));
 }
 
+void trigger_client_reset(const AppSession& app_session)
```

Comment:
Why is this not using the new client reset functionality?

Reply:
I tried out the new `trigger_client_reset` functionality and it does not work in this context. I also had a look at `make_baas_client_reset` which is also quite problematic and would not be as efficient as the old code for this scenario.

Reply:
why doesn't the new trigger_client_reset functionality work in this context?

Reply:
Even though the call to the private API returns a 200 http status code, the before_client_reset handler is never called. I was using the File Ident from the sync session which came directly from the sync manager as the Realm is not available at the time when `on_sync_client_event_hook` is called.